sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -408,7 +408,13 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
   }
 
   private def defaultTablePath(tableIdent: TableIdentifier): String = {
-    val dbLocation = getDatabase(tableIdent.database.get).locationUri
+    // default database's location always use the warehouse path,
+    // and since the location of database stored in metastore is qualified,
+    // here we also make qualify for warehouse location
+    val dbLocation = if (tableIdent.database.orNull == SessionCatalog.DEFAULT_DATABASE) {
+      SessionCatalog.makeQualifiedPath(conf.get(WAREHOUSE_PATH), hadoopConf).toString
+    } else getDatabase(tableIdent.database.get).locationUri
```

Comment:
Please forgive me~
It seemly that I understand your descirbe is for confirm what issue this change is going to resolve . Yes, it is to resolve the issue which you described~